app/services/api/to_user_sessions_hash2.rb

```diff
@@ -0,0 +1,73 @@
+class Api::ToUserSessionsHash2
+  def initialize(form:, user:)
+    @form = form
+    @user = user
+  end
+
+  def call
+    return Failure.new(form.errors) if form.invalid?
+
+    delete_sessions(data.select(&:deleted))
+
+    Success.new(
+      {
+        upload: new_in_mobile_app,
+        download: new_in_db + outdated,
```

Comment:
Have you considered `database` instead of `db`? I find it makes things more clear, helps with autocomplete, grepping and such stuff